src/sagas/index.ts

```diff
@@ -140,23 +140,65 @@ function* workspaceSaga(): SagaIterator {
     }
   })
 
+  /**
+   * Ensures that the external JS libraries have been loaded, with a
+   * timeout of 3 seconds. An error will be thrown if the libraries are not
+   * loaded.
+   */
+  function* checkWebGLAvailable() {
+    /**
+     * A generator function that constantly checks if getReadyWebGLForCanvas
+     * is available in the global (window) scope. Returns (true) if such a function
+     * exists.
+     */
+    function* helper() {
+      while (true) {
+        if ((window as any).getReadyWebGLForCanvas !== undefined) {
+          break
+        }
+        /** Prevent checking instantaenously by waiting for a bit. */
+        yield call(delay, 250)
+      }
+      return true
+    }
+    /** Create a race condition between the js files being loaded and a 3 second timeout. */
+    const { loadedScripts, timeout } = yield race({
+      loadedScripts: call(helper),
+      timeout: call(delay, 4000)
+    })
+    if (timeout !== undefined && loadedScripts === undefined) {
+      yield call(showWarningMessage, 'Error loading libraries', 750)
+      return false
+    } else {
+      return true
+    }
+  }
+
+  /**
+   * Makes a call to checkWebGLAvailable to ensure that the Graphics libraries are loaded.
+   * To abstract this to all libraries, add a call to the all() effect.
+   */
+  yield takeEvery(actionTypes.ENSURE_LIBRARIES_LOADED, function*(action) {
+    yield all([call(checkWebGLAvailable)])
+  })
+
   /**
    * Handles the side effect of resetting the WebGL context when context is reset.
    *
    * @see clearContext and files under 'public/externalLibs/graphics'
    */
   yield takeEvery(actionTypes.CLEAR_CONTEXT, function*(action) {
+    yield all([call(checkWebGLAvailable)])
```

Comment:
Is `all` here to ensure `checkWebGLAvailable` has finished running before this goes to the next line? Doesn't `call` already wait for that? If not, does `yield*` work?

Reply:
Yep, I couldn't get `call` to be blocking but I'll try `yield*`